test/model/typing_status_test.dart

```diff
@@ -0,0 +1,240 @@
+          groupNarrow: {eg.thirdUser.userId: anyTimer}});
+      checkNotifiedOnce();
+
+      checkTyping(topicNarrow, TypingOp.start, eg.selfUser)
+        .typistsByNarrow.deepEquals({
+          dmNarrow: {eg.otherUser.userId: anyTimer},
+          groupNarrow: {eg.thirdUser.userId: anyTimer},
+          topicNarrow: {eg.selfUser.userId: anyTimer}});
+      checkNotifiedOnce();
+    });
+
+    test('add a typist in the same narrow', () {
+      prepareModel();
+
+      checkTyping(groupNarrow, TypingOp.start, eg.selfUser)
+        .typistsByNarrow.deepEquals({
+          groupNarrow: {eg.selfUser.userId: anyTimer}});
+      checkNotifiedOnce();
+
+      checkTyping(groupNarrow, TypingOp.start, eg.thirdUser)
+        .typistsByNarrow.deepEquals({
+          groupNarrow: {eg.selfUser.userId: anyTimer, eg.thirdUser.userId: anyTimer}});
+      checkNotifiedOnce();
+    });
+
+    test('sort dm recipients', () {
+      prepareModel();
+
+      final eventUnsorted = TypingEvent(id: 1, op: TypingOp.start, senderId: eg.selfUser.userId,
+        messageType: MessageType.private,
+        recipientIds: [5, 4, 10, 8, 2, 1, eg.selfUser.userId],
+        streamId: null,
+        topic: null);
+      // DmNarrow's constructor expects the recipient IDs to be sorted.
+      check(() => model.handleTypingEvent(eventUnsorted)).returnsNormally();
+    });
+
+  });
+
+  group('handle typing stop events', () {
+    test('remove a typist from an unknown narrow', () => awaitFakeAsync((async) async {
+      prepareModel();
+
+      checkTyping(groupNarrow, TypingOp.start, eg.otherUser)
+        .typistsByNarrow.deepEquals({groupNarrow: {eg.otherUser.userId: anyTimer}});
+      checkNotifiedOnce();
+
+      checkTyping(dmNarrow, TypingOp.stop, eg.otherUser)
+        .typistsByNarrow.deepEquals({groupNarrow: {eg.otherUser.userId: anyTimer}});
+      checkNotNotified();
+    }));
+
+    test('remove one of two typists in the same narrow', () => awaitFakeAsync((async) async {
+      prepareModel();
+
+      checkTyping(groupNarrow, TypingOp.start, eg.otherUser);
+      checkNotifiedOnce();
+      checkTyping(groupNarrow, TypingOp.start, eg.selfUser);
+      checkNotifiedOnce();
+
+      check(model).typistsByNarrow.deepEquals({
+        groupNarrow: {eg.otherUser.userId: anyTimer, eg.selfUser.userId: anyTimer}});
+
+      checkTyping(groupNarrow, TypingOp.stop, eg.otherUser)
+        .typistsByNarrow.deepEquals({groupNarrow: {eg.selfUser.userId: anyTimer}});
+      checkNotifiedOnce();
+    }));
+
+    test('remove typists from different narrows', () => awaitFakeAsync((async) async {
+      prepareModel();
+
+      checkTyping(dmNarrow, TypingOp.start, eg.otherUser);
+      checkNotifiedOnce();
+      checkTyping(groupNarrow, TypingOp.start, eg.thirdUser);
+      checkNotifiedOnce();
+      checkTyping(topicNarrow, TypingOp.start, eg.selfUser);
+      checkNotifiedOnce();
+
+      async.elapse(const Duration(seconds: 5));
```

Comment:
Does this have a purpose in the story this test is telling? It doesn't seem like it does.